server/channels/app/custom_profile_attributes_test.go

```diff
@@ -0,0 +1,462 @@
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	if cpaGroupID == "" {
+		_, err := th.App.cpaGroupID()
+		require.NoError(t, err)
+	}
+
+	t.Run("should fail if the field is not valid", func(t *testing.T) {
+		field := &model.PropertyField{Name: model.NewId()}
+
+		createdField, err := th.App.CreateCPAField(field)
+		require.NotNil(t, err)
+		require.Empty(t, createdField)
+	})
+
+	t.Run("should not be able to create a property field for a different feature", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: model.NewId(),
+			Name:    model.NewId(),
+			Type:    model.PropertyFieldTypeText,
+		}
+
+		createdField, appErr := th.App.CreateCPAField(field)
+		require.Nil(t, appErr)
+		require.Equal(t, cpaGroupID, createdField.GroupID)
+	})
+
+	t.Run("should correctly create a CPA field", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: cpaGroupID,
+			Name:    model.NewId(),
+			Type:    model.PropertyFieldTypeText,
+			Attrs:   map[string]any{"visibility": "hidden"},
+		}
+
+		createdField, err := th.App.CreateCPAField(field)
+		require.Nil(t, err)
+		require.NotZero(t, createdField.ID)
+		require.Equal(t, cpaGroupID, createdField.GroupID)
+		require.Equal(t, map[string]any{"visibility": "hidden"}, createdField.Attrs)
+
+		fetchedField, gErr := th.App.Srv().propertyService.GetPropertyField(createdField.ID)
+		require.NoError(t, gErr)
+		require.Equal(t, field.Name, fetchedField.Name)
+		require.NotZero(t, fetchedField.CreateAt)
+		require.Equal(t, fetchedField.CreateAt, fetchedField.UpdateAt)
+	})
+
+	t.Run("should not be able to create CPA fields above the limit", func(t *testing.T) {
+		// we create the rest of the fields required to reach the limit
+		for i := 1; i < CustomProfileAttributesFieldLimit-1; i++ {
+			field := &model.PropertyField{
+				Name: model.NewId(),
+				Type: model.PropertyFieldTypeText,
+			}
+			createdField, err := th.App.CreateCPAField(field)
+			require.Nil(t, err)
+			require.NotZero(t, createdField.ID)
+		}
+
+		// then, we create a last one that would exceed the limit
+		field := &model.PropertyField{
+			Name: model.NewId(),
+			Type: model.PropertyFieldTypeText,
+		}
+		createdField, err := th.App.CreateCPAField(field)
+		require.NotNil(t, err)
+		require.Equal(t, http.StatusUnprocessableEntity, err.StatusCode)
+		require.Zero(t, createdField)
+	})
+
+	t.Run("deleted fields should not count for the limit", func(t *testing.T) {
+		// we retrieve the list of fields and check we've reached the limit
+		fields, err := th.App.ListCPAFields()
+		require.Nil(t, err)
+		require.Len(t, fields, CustomProfileAttributesFieldLimit)
+
+		// then we delete one field
+		require.Nil(t, th.App.DeleteCPAField(fields[0].ID))
+
+		// creating a new one should work now
+		field := &model.PropertyField{
+			Name: model.NewId(),
+			Type: model.PropertyFieldTypeText,
+		}
+		createdField, err := th.App.CreateCPAField(field)
+		require.Nil(t, err)
+		require.NotZero(t, createdField.ID)
+	})
+}
```

Comment:
Same here, I feel like tests depending on each other are a bit dangerous. Not urgent for this PR, though, if time is a constraint.

Reply:
Closing this one to centralize the conversation on the previous comment